### PR TITLE
fixes instruments

### PR DESCRIPTION
**Pull request by silicons**
**sigh**

### CODE REVIEW

**Comment by calzilla1** [comment]
*le sigh

**Comment by Carbonhell** [comment]
Don't forget to fix this too: https://github.com/tgstation/tgstation/pull/29395/files#diff-15dcac63c46c718656fde05d909e65b0R76 
should be i, not hearers[i] since that's the associated value ie 1, whereas i is the mob itself

File: code/game/objects/structures/musician.dm [hunk 1]
```diff
@@ -65,15 +65,15 @@
...
 		for(var/mob/M in get_hearers_in_view(15, source))
 			if(!M.client || !(M.client.prefs.toggles & SOUND_INSTRUMENTS))
 				continue
-			LAZYSET(hearers, M, TRUE)
+			LAZYADD(hearing_mobs, M)
 		last_hearcheck = world.time
-	if(LAZYLEN(hearers))
-		for(var/i in hearers)
-			var/mob/M = hearers[i]
+	if(LAZYLEN(hearing_mobs))
```
Comment (RemieRichards) [inline code/game/objects/structures/musician.dm]: this line isn't needed, forloops dont run on empty lists.

**Comment by RemieRichards** [comment]
@Carbonhell No, it shouldn't, just neither should be named ```i```, it should be lowercase ```m```

**Comment by RemieRichards** [comment]
<i>it should be lowercase m</i>

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**